### PR TITLE
cspdk cleanup

### PR DESCRIPTION
**Pull request by flaport**
Some more cleanup of cspdk.

- more consistency over 3 pdk versions
- improved tech (force xs name when default etc)

passes all tests except regression tests (**note: you'll need the current master branch of gdsfactory!**)

Please review and let me know what you think @joamatab

### CODE REVIEW

File: cspdk/sin300/tech.py [hunk 1]
```diff
@@ -1,52 +1,47 @@
...
-    WG: Layer = (3, 0)
-    SLAB: Layer = (5, 0)
-    FLOORPLAN: Layer = (99, 0)
-    HEATER: Layer = (39, 0)
-    GRA: Layer = (6, 0)
-    LBL: Layer = (100, 0)
-    PAD: Layer = (41, 0)
-    NITRIDE: Layer = (203, 0)
-    NITRIDE_ETCH: Layer = (204, 0)
+    # TODO: how can we make this pass type checking?
```
Comment (joamatab) [inline cspdk/sin300/tech.py:26]: remove?

**Review by joamatab — APPROVED** [review]
looks good to me

File: cspdk/sin300/tech.py (hunk 1 above)
Comment (flaport) [reply cspdk/sin300/tech.py]: no, that makes pyright error out (not a huge deal, but I like my repositories type clean)

**Comment by joamatab** [comment]
it's missing the slab on the `die_rc`

image

**Comment by flaport** [comment]
Hi @joamatab , I think everything is now fixed. I went through the yaml definition differences and decided the differences are fine, so I regenerated the yaml test references.

I also went through the gds files and for the most part the differences where fine/to be expected as well. However, there is a minor issue for grating_coupler_rectangular where there seems to be extra padding around the grating in the old version which now seems to be gone in the new version. Can you figure out why? Or if its an issue?

image

**Pull request merged**